src/trace/dma-trace.c

```diff
@@ -291,6 +302,8 @@ static int dma_trace_start(struct dma_trace_data *d)
 	}
 
 	if (d->dc.chan) {
+		unsigned int flags;
```

Comment:
just noticed, that this file uses `unsigned int` for spin-lock flags. It should actually be `uint32_t`. Not something to change in this PR, but a separate PR could update that

Reply:
I'll fix it up.

Reply:
Replaced with `k_spinlock_key_t` as we have switched lock API...